src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.AddProperty.cs

```diff
@@ -28,10 +28,14 @@ private JsonPropertyInfo AddProperty(Type propertyType, PropertyInfo propertyInf
             // Convert interfaces to concrete types.
             if (propertyType.IsInterface && jsonInfo.ClassType == ClassType.Dictionary)
             {
-                Type newPropertyType = jsonInfo.ElementClassInfo.GetPolicyProperty().GetConcreteType(propertyType);
-                if (propertyType != newPropertyType)
+                // If a polymorphic case, we have to wait until run-time values are processed.
+                if (jsonInfo.ElementClassInfo.ClassType != ClassType.Unknown)
```

Comment:
What's the behavior if `jsonInfo.ElementClassInfo.ClassType` **is** `ClassType.Unknown`?

Reply:
Currently a `JsonReaderException` will be thrown later. When we get support for `JsonElement` then I expect that to work (where `object` would signify creating a `JsonElement`).